cdap-ui/app/cdap/components/AbstractWidget/index.tsx

```diff
@@ -14,32 +14,46 @@
  * the License.
 */
 
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
 import AbstractWidgetFactory from 'components/AbstractWidget/AbstractWidgetFactory';
 import StateWrapper from 'components/AbstractWidget/StateWrapper';
 require('./AbstractWidget.scss');
 
-export const WIDGET_PROPTYPES = {
-  widgetProps: PropTypes.object,
-  value: PropTypes.oneOfType([PropTypes.array, PropTypes.string, PropTypes.number]),
-  onChange: PropTypes.func,
-  extraConfig: PropTypes.object,
-};
 export const DEFAULT_WIDGET_PROPS = {
   widgetProps: {},
   value: '',
+  // tslint:disable-next-line:no-empty
   onChange: () => {},
+  disabled: false,
 };
-export default class AbstractWidget extends Component {
-  static propTypes = {
-    type: PropTypes.oneOf(Object.keys(AbstractWidgetFactory)),
-    ...WIDGET_PROPTYPES,
+
+export interface IStageSchema {
+  name: string;
+  schema: string;
+}
+
+export interface IWidgetProps<T = any> {
+  widgetProps: T;
+  value: string | number;
```

Comment:
Should this include the possibility of an array value, to match with WIDGET_PROPS?

Reply:
I'm going to change the WIDGET_PROPS to not allow array. It should just be simple types for now